jdeps: add page (#8758)
tldr-pages/tldr: pages/common/jdeps.md

# jdeps

> > Java Class Dependency Analyzer.

Reviewer: Why the `Title Case`?
```suggestion
> Java class dependency analyzer.
```

Author: I originally did it like that because the `java`, `javac`, `jar`, `jmap` and some other java related commands did the same for those pages. But if title case isn't good I can make that change

Reviewer: I think the `Title Case` doesn't fit here

Reviewer 2: > I think the `Title Case` doesn't fit here

Agreed

Author: I can apply the change here so this page is already altered and then open another PR to change the other java related pages to no longer use title case as well. Does that seem like a good plan?

Reviewer: yep

> > More information: <https://docs.oracle.com/en/java/javase/17/docs/specs/man/jdeps.html>.

Reviewer 2: ```suggestion
> More information: <https://docs.oracle.com/en/java/javase/18/docs/specs/man/jdeps.html>.
```
One last suggestion. Thanks

> - Analyze the dependencies of a jar or class file:

Reviewer 2: ```suggestion
- Analyze the dependencies of a `.jar` or `.class` file:
```

> `jdeps {{filename.class}}`

Reviewer 2: ```suggestion
`jdeps {{path/to/filename.class}}`
```

> - Print a dependency summary:

Reviewer 2: ```suggestion
- Print a summary of all dependencies:
```
I think this would be better. What do you think?

> - Print a summary of all dependencies:

Reviewer: ```suggestion
- Print a summary of all dependencies of a specific Jar file:
```

Reviewer 2: ```suggestion
- Print a summary of all dependencies of a specific `.jar` file:
```
I think this would be better.

> `jdeps {{filename.jar}} -summary`

Reviewer 2: ```suggestion
`jdeps {{path/to/filename.jar}} -summary`
```

> - Print all class-level dependencies:

Reviewer: ```suggestion
- Print all class-level dependencies of a Jar file:
```

Reviewer 2: ```suggestion
- Print all class-level dependencies of a `.jar` file:
```

Would mentioning file extension be better or is the current one fine?

Reviewer: Feels the same for me.

Author: For `.jar` vs `JAR` or `Jar` after running a very quick and simple text search it seems that without the `.` is more common in other pages so I'd lean that way.

> `jdeps {{filename.jar}} -verbose`

Reviewer 2: ```suggestion
`jdeps {{path/to/filename.jar}} -verbose`
```

Author: Ah oops, I'll double check the entire page again quickly then add that suggestion too lol

> - Output the results of the analysis in form of a DOT file:

Reviewer: ```suggestion
- Output the results of the analysis in a DOT file into a specific directory:
```

> `jdeps {{filename.jar}} -dotoutput {{output/path}}`

Reviewer 2: ```suggestion
`jdeps {{path/to/filename.jar}} -dotoutput {{output/path}}`
```

> `jdeps {{path/to/filename.jar}} -dotoutput {{output/path}}`

Reviewer 2: ```suggestion
`jdeps {{path/to/filename.jar}} -dotoutput {{path/to/output}}`
```
I think this sounds better.

> Does this command require a file extension?

Author: It expects the path to a directory to place generated files into, so I was thinking of just changing it to `path/to/directory` which seems to be the standard on other pages.

Author: I made that change along with the other suggestions

Reviewer 2: Thanks

> - Display usage information for the jdeps command:

Reviewer 2: ```suggestion
- Display help:
```

This is the generic command description we use for help command in most pages.
